### PR TITLE
read simple persist-type reader out of vault

### PR DESCRIPTION
**Pull request by deitch**
Any usage of `pkg/pillar/utils`, especially those things calling `WaitFor*()`, ends up depending on `pkg/pillar/containerd`, just to get the containerd client. It then depends on `pkg/pillar/vault`, because it needs to call `vault.ReadPersistType()`.

`ReadPersistType()` is a simple function to read a file. But because it is in `vault`, it in turns depends on the go-zfs library, which is CGO, only available on Linux and with the right packages installed.

That complex dependency shouldn't exist for something like utils. 

This moves just `ReadPersistType()` out of `pkg/pillar/vault` and into `pkg/pillar/utils/persist`. The dependency tree is simplified.

### CODE REVIEW

File: pkg/pillar/utils/persist/persist.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package persist
```
Comment (eriknordmark) [inline pkg/pillar/utils/persist/persist.go:1]: Please add a copyright and license line.

File: pkg/pillar/cas/cas.go [hunk 2]
```diff
@@ -12,28 +12,29 @@ import (
 
 // BlobInfo holds the info of a blob present in CAS's blob store
 type BlobInfo struct {
-	//Digest to identify the blob uniquely. The format will/should be <algo>:<hash> (currently supporting only sha256:<hash>).
+	// Digest to identify the blob uniquely. The format will/should be <algo>:<hash> (currently supporting only sha256:<hash>).
```
Comment (eriknordmark) [inline pkg/pillar/cas/cas.go:15]: If you fix some of these it appears like you need to fix many more in the same functions and structures.

**Review by eriknordmark — APPROVED** [review]
Kick off tests

File: pkg/pillar/cas/cas.go (hunk 2 above)
Comment (deitch) [reply pkg/pillar/cas/cas.go:15]: On the first run, it flagged about 11 of these in the `cas.go` file. I didn't touch it, so wasn't sure why it complained, but I didn't mind an extra commit to fix it. I might have opened a Pandora's Box.

Comment (deitch) [reply pkg/pillar/cas/cas.go:15]: I gave it one more shot to address those. If it gets out of hand, I will remove the commit.

Do you know why it flagged the file, even though I did not touch it?

File: pkg/pillar/utils/persist/persist.go (hunk 1 above)
Comment (deitch) [reply pkg/pillar/utils/persist/persist.go:4]: Done

File: pkg/pillar/cas/cas.go (hunk 2 above)
Comment (deitch) [reply pkg/pillar/cas/cas.go:15]: I addressed a large number of them, all in files I did not go near. I think I have hit my limit on that.

**Review by uncleDecart — APPROVED** [review]
LGTM, Thank you @deitch! (comment linting pandora box is indeed open now :D)

**Comment by deitch** [comment]
If everything passes except for yetus, I will merge it in.

**Pull request merged**